### PR TITLE
FSF-3720 beginning of planing adapters

### PR DESCRIPTION
**Pull request by lourenco31**

### CODE REVIEW

File: src/planning/include/planning/planning.hpp [hunk 1]
```diff
@@ -35,14 +35,16 @@ class Adapter;
  * map topics, and publishing planned path points.
  */
 class Planning : public rclcpp::Node {
-  Mission mission = not_selected;                                /**< Current planning mission */
+  common_lib::competition_logic::Mission mission =
+      common_lib::competition_logic::Mission::NONE;              /**< Current planning mission */
   LocalPathPlanner *local_path_planner = new LocalPathPlanner(); /**< Local path planner instance */
   Adapter *adapter; /**< Adapter instance for external communication */
```
Comment (marhcouto) [inline src/planning/include/planning/planning.hpp:41]: Change to shared ptr

File: src/planning/src/adapter_planning/eufs.cpp [hunk 2]
```diff
@@ -33,4 +43,54 @@ void EufsAdapter::set_mission_state(int mission, int state) {
...
+                         msg.pose.pose.orientation.x * msg.pose.pose.orientation.x -
+                         msg.pose.pose.orientation.y * msg.pose.pose.orientation.y -
+                         msg.pose.pose.orientation.z * msg.pose.pose.orientation.z);
+  this->node->vehicle_localization_callback(pose);
+}
+
 void EufsAdapter::finish() { std::cout << "Finish undefined for Eufs\n"; }
+
+void EufsAdapter::map_callback(const eufs_msgs::msg::ConeArrayWithCovariance& msg) {
+  RCLCPP_INFO(this->node->get_logger(), "Received cones from EUFS");
```
Comment (marhcouto) [inline src/planning/src/adapter_planning/eufs.cpp:63]: Probably not a good idea to keep this as info, very high frequency, will fill the terminal up

File: src/planning/src/adapter_planning/pacsim.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#include "adapter_planning/pacsim.hpp"
+
+#include "planning/planning.hpp"
+
+PacSimAdapter::PacSimAdapter(Planning* planning) : Adapter(planning) {
+  if (this->node->using_simulated_se_) {
+    RCLCPP_DEBUG(this->node->get_logger(), "Using simulated SE in pacsim");
+    tf_buffer_ = std::make_unique<tf2_ros::Buffer>(this->node->get_clock());
+    tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+    auto timer = this->node->create_wall_timer(std::chrono::milliseconds(100),
```
Comment (marhcouto) [inline src/planning/src/adapter_planning/pacsim.cpp:10]: Probably can reduce the milliseconds, the callbacks are usually faster

File: src/planning/src/planning/planning.cpp [hunk 4]
```diff
@@ -36,29 +36,33 @@ Planning::Planning() : Node("planning") {
...
+
+    // Vehicle Localization Subscriber
+    this->vl_sub_ = this->create_subscription<custom_interfaces::msg::VehicleState>(
+        "/state_estimation/vehicle_state", 10,
+        std::bind(&Planning::vehicle_localization_callback, this, _1));
+  }
 }
 
 void Planning::track_map_callback(const custom_interfaces::msg::ConeArray &msg) {
+  RCLCPP_INFO(this->get_logger(), "Planning received %d cones", (int)msg.cone_array.size());
```
Comment (marhcouto) [inline src/planning/src/planning/planning.cpp:65]: Again, INFO might be too high of a logging level for callbacks of high frequency

**Review by marhcouto — CHANGES_REQUESTED** [review]
Small changes requested

File: src/planning/launch/planning.launch.py [hunk 5]
```diff
@@ -73,7 +73,12 @@ def generate_launch_description():
             DeclareLaunchArgument(
                 "adapter",
                 description="Environment to run node on",
-                default_value="fsds",
+                default_value="eufs",
+            ),
+            DeclareLaunchArgument(
+                "use_simulated_se",
+                description="Whether to use simulated state estimation",
+                default_value="1",
```
Comment (andrelds11) [inline src/planning/launch/planning.launch.py:81]: is 1 actually the default value?

File: src/planning/src/planning/planning.cpp [hunk 6]
```diff
@@ -36,29 +36,33 @@ Planning::Planning() : Node("planning") {
...
 
   // Publishes path from file in Skidpad & Acceleration events
   this->timer_ = this->create_wall_timer(
       std::chrono::milliseconds(100), std::bind(&Planning::publish_predicitive_track_points, this));
-
+  RCLCPP_INFO(this->get_logger(), "Using %s adapter", mode.c_str());
   // Adapter to communicate with the car
   this->adapter = adapter_map[mode](this);
+
+  if (!using_simulated_se_ || mode == "vehicle") {
```
[the remark this answers is not in the document]
Comment (lourenco31) [reply src/planning/src/planning/planning.cpp:52]: It's all in the adapters

**Review by marhcouto — APPROVED** [review]
Issues are basically code quality, will be solved later due to urgency in testing

**Pull request merged**